### ISSUE TITLE
Not in focus GUI window causes high CPU utilization

### ISSUE THREAD

**Issue by Hoffs** [microsoft/wslg#226]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.21387.1]
Your Distribution version: Ubuntu 20.04
Your WSLg version: 
WSLg ( x86_64 ): 1.0.22+Branch.main.Sha.d7102a7ff8a3bb04bbcd3483c3f7d7a7633686a1
Mariner: VERSION="1.0.20210224"
FreeRDP: 5f083fa0b97d433d6204985f6047886e29c1c61e
weston: 9e402088aa2cd9316851ea20b9befbcf3b9c9564
pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
mesa:
```

# Steps to reproduce

- Start ubuntu distro
- Install xterm
- Start xterm (doesnt matter if using desktop icon or from within terminal)
- Click to focus the terminal
- Click to focus any other window
- Observe `Vmmem` CPU usage jump

I observed same behavior in Debian + Alacritty/st and Ubuntu + Xterm. On my machine usage jumps from ~0% to ~12%. Once the window is focused again the usage drops to ~0% again.

I also tried changing refresh rate from 144 to 60, but the cpu usage remains the same.

#  Expected behavior

No apparent CPU usage increase when not in focus.

# Actual behavior

Higher CPU usage when out of focus compared to in focus.

When xterm window is focused:
![image](https://user-images.githubusercontent.com/5737899/119230901-dca6c380-bb26-11eb-8d1b-c82a417e15f1.png)
When task manager is focused (snipping tool forces it to lose focus when capturing the image but the point remains the same)
![image](https://user-images.githubusercontent.com/5737899/119230914-f1835700-bb26-11eb-80a1-ac3d9977aaf0.png)

**Comment by onomatopellan** [comment]
Same issue here. When a wayland or x11 app is not in focus my CPU usage jumps to 25%. At that point attaching weston binary to strace shows this repeteadly until it gets the focus again and CPU drops to 0,1%.
```
epoll_wait(5, [{EPOLLIN, {u32=2228879200, u64=94899531284320}}], 32, -1) = 1
recvfrom(27, 0x564f84da0230, 2, 0, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(75, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
```

**Comment by onomatopellan** [comment]
@hideyukn88 Were you able to repro this? After testing every WSLg version released I found this issue happens since WSLg 1.0.21. It looks like something from [this commit](https://github.com/microsoft/weston-mirror/commit/43741db0) is increasing the CPU power usage while a linux app is not focused.

![2021-06-11 (2)](https://user-images.githubusercontent.com/599142/121720012-4e00e300-cae3-11eb-8511-e330a9e62034.png)

**Comment by hideyukn88** [comment]
@onomatopellan , yes, I am able to repro this issue, but it's not by focusing away from xterm window. The way I can repro is by minimizing xterm window, then CPU usage of weston reach 100% in top. I can spend sometime later day to take a look. thanks!

**Comment by mick-p1982** [comment]
FWIW (not interesting) what I saw with a fresh install last week was identical to how it's described. (I.e. once it kicks in, it goes up to ~10% or less until the mouse hovers back over the inactive window. Then it seems to go away.)

**Comment by hideyukn88** [comment]
(memo to myself) somehow file descriptor from FreeRDP's dynamic virtual channel message queue stay readable, thus Wayland display loop keep awake and calling RDP-backend to see any outstanding job, but since that message queue is empty, nothing done. WinPR's MessageQueue_Peek doesn't not reset (= read) fd if queue size is zero, thus fd stay readable. The possibility is some code is making that readable without actually queueing message.

**Comment by onomatopellan** [comment]
I can confirm this issue is fixed with latest weston commit. Thanks a lot!

**Comment by thangamani-arun** [comment]
this happens, after wakeup from hibernation. So be aware of it.  It happens to you, just read here: https://github.com/microsoft/WSL/issues/6982#issuecomment-1500137635

**Comment by hideyukn88** [comment]
@thangamani-arun, would you please share log files from /mnt/wslg while this is happening? thanks!

**Comment by thangamani-arun** [comment]
@hideyukn88 san: here are the logs, but I don't have the issue now. Idon't see any cpu usage at this moment. 

[pulseaudio.log](https://github.com/microsoft/wslg/files/11179975/pulseaudio.log)
[stderr.log](https://github.com/microsoft/wslg/files/11179979/stderr.log)
[weston.log](https://github.com/microsoft/wslg/files/11179980/weston.log)
[wlog.log](https://github.com/microsoft/wslg/files/11179981/wlog.log)

**Comment by hideyukn88** [comment]
@thangamani-arun, thanks for quick response, but I don't see any clue other than below message logged in weston.log. Does this log continue to grow with these messages? thanks!

```
[18:49:13.998] xfixes selection notify event: owner 2097153
[18:49:13.998] our window, skipping
[18:49:14.863] xfixes selection notify event: owner 2097153
[18:49:14.863] our window, skipping
[18:49:40.124] xfixes selection notify event: owner 2097153
[18:49:40.124] our window, skipping
[18:50:23.488] xfixes selection notify event: owner 2097153
```

**Comment by thangamani-arun** [comment]
@hideyukn88 : I'm not sure. but as far as I remembers, When I try to launch Ubuntu-20.04 wsl app, It was keep blinking nothing was happening and it was hung and CPU usage was high. Then after some research I was able to kill and re-launched it without hanging issue.

FYI. I was not able to shutdown the wsl from cmd.exe, the command was infinitely waiting. Hence I had to use `hcsdiag kill` to kill and able to relaunch it.

**Comment by hideyukn88** [comment]
@thangamani-arun, thanks for info, if it happens again, please take a log following the steps at https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-collect-wsl-logs-recommended-method, and share with us, thanks!
